src/aframe_components/custom-components.ts

```diff
@@ -20,24 +25,100 @@ interface State {
 export interface Guidance extends Component {
   state: State;
   pointPositions: Vector3[];
+  kdtree: KDNode | null;
+}
+
+export interface KDNode {
+  position: Vector3;
+  left: KDNode | null;
+  right: KDNode | null;
+}
+
+export function build3DTree(positions: Vector3[], depth: number){
+  const n = positions.length;
+  const mid = Math.floor(n / 2);
+  if (n <= 0){
+    return null;
+  }
+  const axis = depth % 3;
+  positions.sort((a, b) => a[AXIS_DICT[axis]] - b[AXIS_DICT[axis]]);
```

Comment:
Did you give any thought to presorting this array by each dimension ahead of time?